infrastructure/src/test/java/org/corfudb/infrastructure/management/failuredetector/HealingAgentTest.java

```diff
@@ -14,67 +18,105 @@
+        PollReport pollReportMock = getPollReportMock(clusterState);
+        Layout layoutMock = mock(Layout.class);
 
-        when(adviserMock.healedServer(isA(ClusterState.class))).thenReturn(Optional.empty());
-        CompletableFuture<DetectorTask> skipped = agent.detectHealing(mock(PollReport.class), mock(Layout.class));
+        CompletableFuture<Boolean> handle = new CompletableFuture<>();
+        handle.completeExceptionally(new FailureDetectorException("err"));
+        doReturn(handle)
+                .when(agentSpy)
+                .handleHealing(same(pollReportMock), same(layoutMock), anySet());
 
-        assertEquals(DetectorTask.SKIPPED, skipped.join());
-    }
+        CompletableFuture<DetectorTask> healingFailed = agentSpy.detectAndHandleHealing(pollReportMock, layoutMock);
 
-    @Test
-    void detectHealingAFailedNode() {
-        ClusterAdvisor adviserMock = mock(ClusterAdvisor.class);
-        NodeRank failedNode = new NodeRank(NodeNames.A, 3);
-        when(adviserMock.healedServer(any(ClusterState.class))).thenReturn(Optional.of(failedNode));
+        verify(agentSpy, times(1))
+                .handleHealing(same(pollReportMock), same(layoutMock), anySet());
 
-        HealingAgent agent = HealingAgent.builder()
-                .dataStore(mock(FailureDetectorDataStore.class))
-                .advisor(adviserMock)
-                .failureDetectorWorker(mock(ExecutorService.class))
-                .localEndpoint(NodeNames.A)
-                .runtimeSingleton(mock(SingletonResource.class))
-                .build();
+        assertEquals(DetectorTask.NOT_COMPLETED, healingFailed.join());
+    }
 
-        HealingAgent agentSpy = spy(agent);
+    @Test
+    void detectAndHandleHealingAFullyConnectedNode() {
+        final String localEndpoint = NodeNames.C;
+        final long epoch = 0;
 
-        PollReport pollReportMock = mock(PollReport.class);
-        when(pollReportMock.getClusterState()).thenReturn(mock(ClusterState.class));
+        HealingAgent agentSpy = getHealingAgentSpy(localEndpoint);
+        ClusterState clusterState = getClusterState(localEndpoint, epoch);
+        PollReport pollReportMock = getPollReportMock(clusterState);
+        Layout layoutMock = mock(Layout.class);
 
-        CompletableFuture<Boolean> handle = new CompletableFuture<>();
-        handle.completeExceptionally(new FailureDetectorException("err"));
+        CompletableFuture<Boolean> handle = CompletableFuture.completedFuture(true);
         doReturn(handle)
                 .when(agentSpy)
-                .handleHealing(any(PollReport.class), any(Layout.class), anySet());
+                .handleHealing(same(pollReportMock), same(layoutMock), anySet());
 
-        CompletableFuture<DetectorTask> healingFailed = agentSpy.detectHealing(pollReportMock, mock(Layout.class));
+        CompletableFuture<DetectorTask> healingFailed = agentSpy.detectAndHandleHealing(pollReportMock, layoutMock);
```

Comment:
Consider renaming this since the healing has not failed.

Reply:
What name would you suggest?